### PR TITLE
ROC-5811:Use next working day api

### PR DESCRIPTION
**Pull request by KiranVarma02**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-5811

### Change description
Calendar API to get next working day

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk [hunk 1]
```diff
@@ -144,14 +144,16 @@
...
     <h2 class="heading-small">{{ t('The defendant’s response') }}</h2>
     <p>{{ t('{{ defendantName }} has rejected the claim. They’ve suggested mediation to help resolve this dispute.',
         { defendantName: claim.claimData.defendant.name }) }}</p>
     <p>{{ t(externalLink('Find out how mediation works', 'https://formfinder.hmctsformfinder.justice.gov.uk/ex730-eng.pdf') + ' (PDF, 399KB)') | safe }}</p>
-    <p>{{ t('You need to email {{ emailAddress }} before before 4pm on {{ respondToMediationDeadline }} to tell us if you agree to mediation or not.',
-        { emailAddress: supportEmailAddress, respondToMediationDeadline: claim.respondToMediationDeadline | date }) }}</p>
+    {% if mediationDeadline %}
+    <p>{{ t('You need to email {{ emailAddress }} before 4pm on {{ respondToMediationDeadline }} to tell us if you agree to mediation or not.',
+        { emailAddress: supportEmailAddress, respondToMediationDeadline: mediationDeadline | date }) }}</p>
+    {% endif %}
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk]: this if ends too soon. The next line should also be included in the if statement

Comment (KiranVarma02) [reply src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk]: agree, done

**Comment by KiranVarma02** [comment]
> I will let me get caught up to speed on the project
> […](#)
> On Mon, Jul 22, 2019, 10:47 Kiran Varma ***@***.***> wrote: addressing @rob review comments -- You are receiving this because you were mentioned. Reply to this email directly or view it on GitHub: [d5550cc](https://github.com/hmcts/cmc-citizen-frontend/commit/d5550cc08e6c361abda56af0de275517a5ac5c77)

Please pick up your speed ;)

File: src/test/app/claims/models/claim.ts [hunk 2]
```diff
@@ -672,16 +672,10 @@ describe('Claim', () => {
 
   describe('respondToMediationDeadline', () => {
 
-    it('should add 5 days to the response deadline', () => {
+    it('should return undefined if claim is not responded to', async () => {
```
Comment (andrewwa-kainos) [inline src/test/app/claims/models/claim.ts]: Please include a test to verify it invokes the `CalendarClient` with 5 days and passes on its response

File: src/test/app/claims/models/claim.ts [hunk 3]
```diff
@@ -672,16 +672,10 @@ describe('Claim', () => {
...
-      claim.respondedAt = moment()
-
-      expect(claim.respondToMediationDeadline.toISOString()).to.equal(claim.respondedAt.add(5, 'days').toISOString())
-    })
-
-    it('should return undefined if claim is not responded to', () => {
-      const claim = new Claim()
-      expect(claim.respondToMediationDeadline).to.equal(undefined)
+      const mediationDeadline = await claim.respondToMediationDeadline()
+      expect(mediationDeadline).to.equal(undefined)
```
Comment (andrewwa-kainos) [inline src/test/app/claims/models/claim.ts]: Can be `.to.be.undefined`

File: src/test/app/claims/calendarClient.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import { mockNextWorkingDay, rejectNextWorkingDay } from 'test/http-mocks/claim-store'
+
+describe('calendar', () => {
+  const SATURDAY = MomentFactory.parse('2019-06-29')
+  const MONDAY_AFTER = '2019-06-29'
+  let calendarClient: CalendarClient = new CalendarClient()
+
+  beforeEach(() => {
+    calendarClient = new CalendarClient()
+  })
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply src/test/app/claims/calendarClient.ts]: agree, done

File: src/test/app/claims/calendarClient.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    const nextWorkingDay = await calendarClient.getNextWorkingDay(SATURDAY, 5)
+
+    expect(nextWorkingDay.toISOString()).to.equal(MomentFactory.parse(MONDAY_AFTER).toISOString())
+  })
+
+  it('should return Unable to get next working day bad request', () => {
+    rejectNextWorkingDay(SATURDAY)
+
+    return calendarClient.getNextWorkingDay(SATURDAY, 0)
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply src/test/app/claims/calendarClient.ts]: agree, done

File: src/test/app/claims/models/claim.ts (hunk 2 above)
Comment (KiranVarma02) [reply src/test/app/claims/models/claim.ts]: done

File: src/test/app/claims/models/claim.ts (hunk 3 above)
Comment (KiranVarma02) [reply src/test/app/claims/models/claim.ts]: done

File: src/test/app/claims/calendarClient.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply src/test/app/claims/calendarClient.ts]: done

**Pull request merged**